src/Search/Annotation/GearmanTaskDriver.php

```diff
@@ -58,16 +60,17 @@ public function registerWorkflow(Workflow $workflow)
         }
     }
 
-    public function addTasksToWorker(GearmanWorker $worker, LoggerInterface $logger)
+    public function addTasksToWorker(GearmanWorker $worker)
     {
         foreach ($this->tasks as $task) {
-            $this->addTaskToWorker($task, $worker, $logger);
+            $this->addTaskToWorker($task, $worker);
         }
     }
 
-    public function addTaskToWorker(GearmanTaskInstance $task, GearmanWorker $worker, LoggerInterface $logger)
+    public function addTaskToWorker(GearmanTaskInstance $task, GearmanWorker $worker)
     {
-        $worker->addFunction($task->name, Closure::bind(function (GearmanJob $job) use ($task, $logger) {
+        $worker->addFunction($task->name, Closure::bind(function (GearmanJob $job) use ($task) {
+            $this->logger->info('GearmanTaskDriver task started', ['task' => $task->name]);
```

Comment:
Under normal circumstances this part of the code should not fail at run time (memory issues aside) the errors should be caught in the workflow methods that are called lines 82/84.

Theres no real way to get accurate id/type without knowing more about which task is running and the format of the data/response. I think it would be best to keep that logic contained within the workflows. If there is an exception thrown in this code we could catch the exception, deserialize (using the workflow) and perform a best guess at the data, or just dump the raw data. That might help identify